src/tests/ftest/io/dfuse_sparse_file.py

```diff
@@ -0,0 +1,172 @@
+                      get_remote_file_size(
+                          self.hostlist_clients[0], sparse_file))
+
+        # create and open a connection on remote node to open file on that
+        # remote node
+        ssh = paramiko.SSHClient()
+        ssh.load_system_host_keys()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(self.hostlist_clients[0], username=getuser())
+        sftp = ssh.open_sftp()
+
+        # open remote file
+        file_obj = sftp.open(sparse_file, 'r+')
+        # set file size to max available nvme size
+        sftp.truncate(sparse_file, self.space_before)
+        self.log.info("File size (in bytes) after truncate: %s",
+                      get_remote_file_size(
+                          self.hostlist_clients[0], sparse_file))
+        # verifying the file size got set to desired value
+        self.assertTrue(
+            get_remote_file_size(
+                self.hostlist_clients[0], sparse_file) == self.space_before)
+
+        # write to the first byte of the file with char 'A'
+        dd_first_byte = u"echo 'A' | dd conv=notrunc of={} bs=1 count=1".\
+                        format(sparse_file)
+        self.execute_cmd(dd_first_byte)
+        self.log.info("File size (in bytes) after writing first byte: %s",
+                      get_remote_file_size(
+                          self.hostlist_clients[0], sparse_file))
+        # verify file did not got overriten after dd write.
+        self.assertTrue(
+            get_remote_file_size(
+                self.hostlist_clients[0], sparse_file) == self.space_before)
+
+        # write to the 1024th byte position of the file
+        dd_1024_byte = u"echo 'A' | dd conv=notrunc of={} obs=1 seek=1023 \
+                       bs=1 count=1".format(sparse_file)
+        self.execute_cmd(dd_1024_byte)
+        self.log.info("File size (in bytes) after writing first byte: %s",
+                      get_remote_file_size(
+                          self.hostlist_clients[0], sparse_file))
+        # verify file did not got overriten after dd write.
+        self.assertTrue(
+            get_remote_file_size(
+                self.hostlist_clients[0], sparse_file) == self.space_before)
+
+        # Obtainer the value of 1st byte and 1024th bytes in the file and
+        # compare their values, they should be same.
+        check_first_byte = file_obj.read(1)
+        file_obj.seek(1022, 1)
+        check_1024th_byte = file_obj.read(1)
+        self.assertTrue(check_first_byte == check_1024th_byte)
+
+        # check the middle 1022 bytes if they are filled with zeros
+        middle_1022_bytes = u"cmp --ignore-initial=1 --bytes=1022 {} {}".\
+                                  format(sparse_file, "/dev/zero")
+        self.execute_cmd(middle_1022_bytes)
+
+        # read last 512 bytes which should be zeros till end of file.
+        ignore_bytes = self.space_before - 512
```

Comment:
Why are we expecting the cmp to fail ? The last 512 bytes need to be zero right ? If we try to read beyond 512, we should expect it to fail.

Reply:
This is one of the original requirements of the ticket to verify and read holes and reach EOF